### ISSUE TITLE
Gutenframe: Consider pulling denied plugins from an endpoint [8]

### ISSUE THREAD

**Issue by kwight** [Automattic/wp-calypso#32959]
In https://github.com/Automattic/wp-calypso/pull/32915, we added the ability to deny Gutenframe to sites that have any plugins within a defined list active. If we were to pull this data from an endpoint, support staff could add to the list through their own tools, giving a more dynamic list with quicker updates.

**Comment by kwight** [comment]
We have D28160-code and D28205-code in progress for handling this. See also pMz3w-9Qt-p2 regarding data persistence.

**Comment by kwight** [comment]
Regarding data persistance, we decided to store the denied plugins list as an array on the `gutenframe_denied_plugins` site option. When a change is logged in Kibana, we also save a full copy of the new list on the `extra` property. If the size grows to over 2K, we don't update and then warn Serenity in Slack. We also log to Slack if the site option ever gets removed or emptied for some reason.

**Comment by mmtr** [comment]
We finally merged the last piece of this work 🎉 

To recap:
- D28205-code: Added a new library to manage the denied plugins, which are now stored as an array on the `gutenframe_denied_plugins` site option (polished in D28608-code, D28667-code and D28708-code).
- D28610-code:  Added another library to determine in the server the editing flow we should load in Calypso (which also checks if the site has any denied plugin active).
- D28605-code: New tool in the MC for allowing HEs to manage the denied plugins list using the new library added in D28205-code (improved in D28681-code and D28712).
- D28222-code: Updated the `/sites/:site/gutenberg` endpoint to include the editing flow determined by the library added in D28610-code.
- https://github.com/Automattic/wp-calypso/pull/33221: We removed from Calypso the logic that was determining the editing flow in favor of using the field of the updated `/sites/:site/gutenberg` endpoint.
- D28344-code: Updated the blog RC in MC with a card that indicates if the site has any denied plugin active where HEs can add new plugins to the list.

It took 10 days of work of 2 devs (20 days in total).
